File: call-in-script/variables.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import os
import subprocess

os.putenv("SAC_DISPLAY_COPYRIGHT", "0")

p = subprocess.Popen(['sac'], stdin=subprocess.PIPE)

s = ""
s += "fg seismo \n"
s += "ch kuser0 &1,kevnm& \n"
tmp = "ABC"   # Python变量，用于代替SAC中的黑板变量
s += "ch kuser1 %s \n" % (tmp)
s += "lh kuser0 kuser1 \n"
s += "q \n"

p.communicate(s.encode())

File: call-in-script/variables.pl
#!/usr/bin/env perl
use strict;
use warnings;
$ENV{SAC_DISPLAY_COPYRIGHT}=0;

open(SAC, "| sac ") or die "Error opening sac\n";
print SAC "fg seismo \n";
print SAC "ch kuser0 &1,kevnm& \n";
my $tmp = "ABC";                # Perl中的变量
print SAC "ch kuser1 $tmp \n";
print SAC "lh kuser0 kuser1 \n";
print SAC "quit \n";
close(SAC);
